src/luascript.cpp

```diff
@@ -12608,6 +13360,383 @@ int LuaScriptInterface::luaPartySetSharedExperience(lua_State* L)
 	return 1;
 }
 
+// Loot
+
+int LuaScriptInterface::luaCreateLoot(lua_State* L)
+{
+	// Loot() will create a new loot item
+	Loot* loot = new Loot();
```

Comment:
Cloned your repo, added a counter in the Loot and MonsterSpell constructors and destructors. Ran 10000 reloads. No cleanup occured and over 20 MiB were leaked. You didn't notice it because you only leak 2168 bytes per reload. As I've said the GC cannot reason about your intents and will not cleanup memory allocated by the C++ code, unless you explicitly provide it with facilities to do so (in this case, by registering the __gc method).